exporter/exporterhelper/internal/persistent_queue.go

```diff
@@ -21,53 +20,32 @@ var (
 // persistentQueue holds the queue backed by file storage
 type persistentQueue struct {
 	*persistentContiguousStorage
-	stopWG       sync.WaitGroup
-	set          exporter.CreateSettings
-	storageID    component.ID
-	numConsumers int
+	set       exporter.CreateSettings
+	storageID component.ID
+	dataType  component.DataType
 }
 
 // NewPersistentQueue creates a new queue backed by file storage; name and signal must be a unique combination that identifies the queue storage
-func NewPersistentQueue(capacity int, numConsumers int, storageID component.ID, marshaler QueueRequestMarshaler,
-	unmarshaler QueueRequestUnmarshaler, set exporter.CreateSettings) Queue {
+func NewPersistentQueue(capacity int, dataType component.DataType, storageID component.ID,
+	marshaler QueueRequestMarshaler, unmarshaler QueueRequestUnmarshaler, set exporter.CreateSettings) Queue {
 	return &persistentQueue{
 		persistentContiguousStorage: newPersistentContiguousStorage(set.Logger, uint64(capacity), marshaler, unmarshaler),
-		numConsumers:                numConsumers,
 		set:                         set,
 		storageID:                   storageID,
+		dataType:                    dataType,
 	}
 }
 
 // Start starts the persistentQueue with the given number of consumers.
-func (pq *persistentQueue) Start(ctx context.Context, host component.Host, set QueueSettings) error {
-	storageClient, err := toStorageClient(ctx, pq.storageID, host, pq.set.ID, set.DataType)
+func (pq *persistentQueue) Start(ctx context.Context, host component.Host) error {
+	storageClient, err := toStorageClient(ctx, pq.storageID, host, pq.set.ID, pq.dataType)
```

Comment:
Should move this into the `pq.persistentContiguousStorage.start` later and remove `persistentQueue`

Reply:
I was thinking about it as well 👍